vertical-pod-autoscaler/pkg/recommender/main.go

```diff
@@ -116,6 +119,9 @@ const (
 )
 
 func main() {
+	flag.Var(&maxAllowedCPU, "max-allowed-cpu", "Maximum amount of CPU that will be recommended for a container. VerticalPodAutoscaler-level maximum allowed takes precedence over the global maximum allowed.")
+	flag.Var(&maxAllowedMemory, "max-allowed-memory", "Maximum amount of memory that will be recommended for a container. VerticalPodAutoscaler-level maximum allowed takes precedence over the global maximum allowed.")
```

Comment:
Should these flags live in `pkg/recommender/logic/recommender.go`, alongside `pod-recommendation-min-memory-mb` and `pod-recommendation-min-cpu-millicores` ?

Also, should they have similar names to the min flags, such as: `container-recommendation-max-memory-mb` and `container-recommendation-max-cpu-millicores` ?

Reply:
In https://github.com/kubernetes/autoscaler/issues/7147#issuecomment-2501042262, I already tried to explain why  Pod-level flags like `--container-recommendation-max-memory-mb`/`--container-recommendation-max-cpu-millicores` don't make sense for max allowed:
> On the other side, recommendations in VPA are on container-level. VPA can be enabled for > 1 container in a Pod. The question is how VPA is expected to distribute the Pod-level max allowed recommendation across the many containers in the Pod. Some approaches are - distribute the configured max allowed recommendation values evenly; distribute proportionally (proportionally based on what?). My conclusion is that VPA should not do that as it does not know about the workload.

IMO the existing `--container-recommendation-max-memory-mb`/`--container-recommendation-max-cpu-millicores` are not okay. They are on Pod-level and the vpa-recommender splits them evenly to the number of containers in the Pod. The implementation of these flags is such that it overwrites the `uncappedTarget` which is undesired according to the definition of the `uncappedTarget` recommendation.

Reply:
I'm not quite sure I understand your response, but my comments weren't very clear. 
I'm not suggesting changing to Pod-level maximum flags.

1. Does it make sense to move these flags into `pkg/recommender/logic/recommender.go`, so that all the max/min flags live near each other in the code base.
2. Does it make sense to rename the `max-allowed-cpu` and `max-allowed-memory` flags (that are being introduced in this PR) to `container-recommendation-max-memory-mb` and `container-recommendation-min-cpu-millicores`, similar to what you described in https://github.com/kubernetes/autoscaler/issues/7147#issuecomment-2501042262

Sorry for misunderstanding, I hope that clears it up?

Reply:
Yes, it makes sense to prefix the flags with `--container-recommendation` to make the purpose of the flags clear.
I addressed the suggestion in https://github.com/kubernetes/autoscaler/pull/7560/commits/7e427ae770fbe13b53c80e188c5c9d55a4cd2ec3.